src/libs/actions/PersistedRequests.ts

```diff
@@ -17,10 +18,61 @@ function clear() {
     return Onyx.set(ONYXKEYS.PERSISTED_REQUESTS, []);
 }
 
+/**
+ * Method to merge two arrays of OnyxUpdate elements.
+ * Elements from the old array which keys are not present in the new array are merged with the data of the new array.
+ */
+function mergeOnyxUpdateData(oldData: OnyxUpdate[] = [], newData: OnyxUpdate[] = []): OnyxUpdate[] {
+    const mergedData = [...newData];
+
+    oldData.forEach((oldUpdate) => {
+        const hasSameKey = newData.some((newUpdate) => newUpdate.key === oldUpdate.key);
+
+        if (!hasSameKey) {
+            mergedData.push(oldUpdate);
+        }
+    });
+
+    return mergedData;
+}
+
+function createUpdatedRequest(oldRequest: Request, newRequest: Request): Request {
+    /**
+     * In order to create updated request, properties: data, failureData, successData and optimisticData have to be merged
+     */
+    const updatedRequest = {
+        data: lodashMerge({...oldRequest.data}, newRequest.data),
+        failureData: mergeOnyxUpdateData(oldRequest.failureData, newRequest.failureData),
+        successData: mergeOnyxUpdateData(oldRequest.successData, newRequest.successData),
+        ...newRequest,
+    };
+
+    const updatedOptimisticData = mergeOnyxUpdateData(oldRequest.optimisticData, newRequest.optimisticData);
+
+    if (updatedOptimisticData.length > 0) {
+        updatedRequest.optimisticData = updatedOptimisticData;
+    }
+
+    return updatedRequest;
+}
+
 function save(requestsToPersist: Request[]) {
```

Comment:
Noticing that this function is only used once, and it's only ever provided a single-item array. We should change it's signature accordingly to simplify things here